appinventor/blocklyeditor/src/blocks/text.js

```diff
@@ -612,26 +609,19 @@ Blockly.Blocks['obfuscated_text'] = {
     this.setColour(Blockly.TEXT_CATEGORY_HUE);
     var label = Blockly.Msg.LANG_TEXT_TEXT_OBFUSCATE + " " +
         Blockly.Msg.LANG_TEXT_TEXT_LEFT_QUOTE
-    var textInput = new Blockly.FieldTextBlockInput('');
+    var textInput = new Blockly.FieldTextInput('');
     textInput.onFinishEditing_ = Blockly.Blocks.text
-        .bumpBlockOnFinishEdit.bind(this);
+        .setOutputOnFinishEdit.bind(this);
     this.appendDummyInput()
         .appendField(label)
         .appendField(textInput,'TEXT')
         .appendField(Blockly.Msg.LANG_TEXT_TEXT_RIGHT_QUOTE);
-    this.setOutput(true, [Blockly.Blocks.text.connectionCheck]);
+    this.setOutput(true, AI.BlockUtils.YailTypeToBlocklyType("text", AI.BlockUtils.OUTPUT));
     this.setTooltip(Blockly.Msg.LANG_TEXT_TEXT_OBFUSCATE_TOOLTIP);
     this.confounder = Math.random().toString(36).replace(/[^a-z]+/g, '').substr(0, 8);
   },
-  domToMutation: function(xmlElement) {
-    var confounder = xmlElement.getAttribute('confounder');
-    this.confounder = confounder;
-  },
-  mutationToDom: function() {
-    var container = document.createElement('mutation')
-    container.setAttribute('confounder', this.confounder);
-    return container;
-  },
+  domToMutation: Blockly.Blocks.text.domToMutation,
+  mutationToDom: Blockly.Blocks.text.mutationToDom,
```

Comment:
Need to double check that this is correct.

Reply:
IIRC, my thinking on this was that since, as far as I could tell, the `confounder` (which used to be added to the mutators) was set in the `init` function and never changed, that it didn't really need to be in the mutators.   But maybe I was missing something.

Reply:
I think it just comes down to reproducible builds. The confounder is used in the YAIL code generation to obfuscate the text. If we don't save it in the mutation, then every time the user a loads a project they will get a different confounder. I don't think it matters either way but to keep the behavior of the system as close to the existing behavior I've restored the old logic and combined it with the calls to the `text` block versions to deal with the output mutation (although I have questions about that too).

Reply:
Got it.  Makes sense.  I didn't quite realize that the confounder changes on each load.